controllers/result.go

```diff
@@ -73,8 +72,17 @@ func CreateResult(ctx context.Context, w http.ResponseWriter, r *http.Request) (
 	}
 
 	if len(resultKeys) == 1 {
-		json.NewEncoder(w).Encode(results[0].Key(resultKeys[0]))
-		return http.StatusOK, nil
+		var u model.User
+		if err = datastore.Get(ctx, p.User, &u); err != nil {
+			return http.StatusInternalServerError, nil
+		}
+
+		if results[0].Finished.Equal(time.Time{}) || u.Company != nil {
```

Comment:
`time.Time{}` is the zero value of `time.Time`, e.g. "no value" or the default value, so this can be explained as: `results[0].Finished` is not set